File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        if(intervals.size() == 1) return intervals;
        sort(intervals.begin(), intervals.end());

        vector<vector<int>> merged;
        merged.push_back(intervals[0]);
        int prevInd;

        for(int i=1;i<intervals.size();i++)
        {   
            prevInd = merged.back()[1];
            if( prevInd >= intervals[i][0])
            {
                merged.back()[1] = max(prevInd, intervals[i][1]);
            }
            else
            {
                merged.push_back(intervals[i]);
            }
        }

        return merged;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        
        if(len(intervals) == 1):
            return intervals

        intervals.sort()

        op = []
        op.append(intervals[0])

        prevEnd = op[0][1]
        
        for i in range(1, len(intervals)):
            currStart = intervals[i][0]
            currEnd = intervals[i][1]
            prevEnd = op[-1][1]

            if prevEnd >= currStart:
                op[-1][1] = max(prevEnd, currEnd)
            else:
                op.append(intervals[i])

        return op